src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -568,6 +568,33 @@ void removeUnfundedOffers (ApplyView& view, std::vector<uint256> const& offers,
     }
 }
 
+void
+Transactor::claimFee (XRPAmount& fee, TER terResult, std::vector<uint256> const& removedOffers)
+{
+    ctx_.discard();
+
+    auto const txnAcct = view().peek(
+        keylet::account(ctx_.tx.getAccountID(sfAccount)));
+
+    auto const balance = txnAcct->getFieldAmount (sfBalance).xrp ();
+
+    // balance should have already been
+    // checked in checkFee / preFlight.
+    assert(balance != zero && (!view().open() || balance >= fee));
+    // We retry/reject the transaction if the account
+    // balance is zero or we're applying against an open
+    // ledger and the balance is less than the fee
+    if (fee > balance)
+        fee = balance;
+    txnAcct->setFieldAmount (sfBalance, balance - fee);
+    txnAcct->setFieldU32 (sfSequence, ctx_.tx.getSequence() + 1);
+
+    if (terResult == tecOVERSIZE && (! removedOffers.empty()))
```

Comment:
that was existing code (refactored) - but I agree, I don't see much benefit in that empty check.